### PR TITLE
fixes to get istiod working with constrained permissions

### PR DESCRIPTION
**Pull request by rcernich**
Signed-off-by: rcernich <rcernich@redhat.com>

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go [hunk 1]
```diff
@@ -132,7 +131,21 @@ func NewNamespaceController(data func() map[string]string, options Options, kube
 	})
 	c.configMapController = configmapInformer
 
-	namespaceInformer := informer.NewNamespaceInformer(kubeClient, options.ResyncPeriod, cache.Indexers{})
+	namespacemlw := listwatch.MultiNamespaceListerWatcher(watchedNamespaceList, func(namespace string) cache.ListerWatcher {
```
Comment (dgn) [inline pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go:134]: This won't work as namespaces are cluster-scoped. I think we'll need to remove the watching of namespaces entirely

Comment (dgn) [reply pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go:134]: Looks like this is used for watching namespaces and creating/updating a configMap with a certificate. We should probably move that into the operator. I filed https://issues.redhat.com/browse/MAISTRA-1724 to take care of it

**Review by dgn — COMMENTED** [review]
I think we should split this into two PRs as it's covering two separate permissions; nodes and namespaces. Namespaces are a bit more complicated (see my inline comment). The part about the nodeInformer should be straightforward as we have that functionality in the operator already.

I created https://issues.redhat.com/browse/MAISTRA-1722 to track any permission issues we find. I think there's no rush for now; these can be fixed in later sprints if need be

**Comment by bison** [comment]
I called out one of the Node informers after the rebase. More info on that in [MAISTRA-1622](https://issues.redhat.com/browse/MAISTRA-1622). I think that change is partly to support exposing gateways via NodePort. Maybe we don't support that, but we should probably document that.

File: pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go [hunk 2]
```diff
@@ -76,7 +75,7 @@ func NewNamespaceController(data func() map[string]string, options Options, kube
 
 	watchedNamespaceList := strings.Split(options.WatchedNamespaces, ",")
 
-	mlw := listwatch.MultiNamespaceListerWatcher(watchedNamespaceList, func(namespace string) cache.ListerWatcher {
+	configmapmlw := listwatch.MultiNamespaceListerWatcher(watchedNamespaceList, func(namespace string) cache.ListerWatcher {
```
Comment (bison) [inline pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go:78]: This seems unrelated.

File: pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go (hunk 1 above)
Comment (rcernich) [reply pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go:134]: Yeah, I figured this was pretty cheesy.  I think we had something like this in 1.1 though, although maybe mlw worked differently there (that's where I pulled it from).  I was considering adding a listener to the member roll controller, but that required a bit more work to hook it in. (Basically, have the listener add events to the queue for each namespace in the list.)

File: pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go (hunk 2 above)
Comment (rcernich) [reply pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go:78]: Because I added another mlw, I simply made the name more descriptive.

**Comment by rcernich** [comment]
Yeah, I'm not sure why it's adding the node IPs for the gateways into the service registry.  Maybe this is for multicluster?  If so, maybe we should be watching routes instead?  (Although, that probably won't work, as you can't access the service using an IP, without specifying SNI host too, and that means we need to get the host IP for the router anyway.)

File: pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go (hunk 2 above)
Comment (bison) [reply pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go:78]: Sure. I was mostly thinking that it was an unnecessary diff to carry, but this has all vanished upstream anyway.

File: pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go (hunk 1 above)
Comment (dgn) [reply pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go:134]: Yeah, that probably makes the most sense - use the NamespaceUpdated events from the MRC and pipe it into this ConfigMap-creating code
